lib/ex_unit/lib/ex_unit/filters.ex

```diff
@@ -27,22 +27,13 @@ defmodule ExUnit.Filters do
       [part] ->
         {part, []}
 
-      parts ->
-        {reversed_line_numbers, reversed_path_parts} =
-          parts
-          |> Enum.reverse()
-          |> Enum.split_while(&match?({_, ""}, Integer.parse(&1)))
-
-        line_numbers =
-          for line_number <- reversed_line_numbers,
-              valid_line_number?(line_number),
-              reduce: [],
-              do: (acc -> [line: line_number] ++ acc)
-
-        path =
-          reversed_path_parts
-          |> Enum.reverse()
-          |> Enum.join(":")
+      [path | parts] ->
+        {path_parts, line_numbers} =
+          parts |> Enum.split_while(&(not match?({_, ""}, Integer.parse(&1))))
```

Comment:
Nitpick: we don't tend to pipe once. :)

```suggestion
          Enum.split_while(parts, &(not match?({_, ""}, Integer.parse(&1))))
```

Reply:
I also corrected the `Enum.join()`...